src/CommandingOfficers/AW4/RuinedCommander.java

```diff
@@ -116,10 +117,20 @@ public ArrayList<CommanderAbility> getReadyAbilities()
     return new ArrayList<CommanderAbility>();
   }
   @Override
-  public void onCOULost(Unit minion)
+  public GameEventQueue onCOULost(Unit minion)
   {
     modifyAbilityStars(-42);
     zoneIsGlobal = false;
+
+    // Stolen from Commander.revertActiveAbility(), with the caveat that we know all our subclass's powers have no revert events.
```

Comment:
I had to add this because Waylon's powers specifically do actually matter while his COU is dead.
It's kind of icky that it doesn't use the main revert function, but I didn't feel like touching the declaration of `GameEvent.sendToListener()` to daisy-chain the master map down.

Reply:
You mention doing this for Waylon specifically, but this is in the base class. Will it not also cause other CO's abilities to persist after COU death?

Reply:
The correct behavior for DoR COs is "if the COU is killed during COP, immediately end the power".
Most of this is fulfilled by "simply revoke the global zone", since move boosts/instant healing/etc don't matter much when it's not your turn (and you obviously can't charge with no zone).

Waylon COP adds combat stats, which apply whether or not his zone is in the picture.
I guess I could add a specific exception there, but that seems a lot less clean than "just end the power".
The compromise here is there's no nice way to get at a `MapMaster` instance, so we can't get revert events for the power... but none supply any. (We can just steal it from the CO's `MapPerspective`, but I like to pretend that's not an option because spaghetti.)

Reply:
> The correct behavior for DoR COs is "if the COU is killed during COP, immediately end the power"

Ok, sure. when COU dies, revert the power. Simple. But you also say.

> Waylon's powers specifically do actually matter while his COU is dead

Except it appears this unconditionally reverts powers on COU death (including Wingman's `moveMod` that grants defense). I don't see the difference between Waylon's ability and say, Gage's (whose `moveMod` grants indirect range - this appears to be a common unit modifier variable name regardless of effect).

I feel like I'm missing something - how does this treat Waylon differently? Why would his powers outlive his COU when others don't?

Reply:
> > Waylon's powers specifically do actually matter while his COU is dead
> 
> Except it appears this unconditionally reverts powers on COU death (including Wingman's `moveMod` that grants defense).

Yes. It does that because otherwise Waylon's ability *would* still work after he dies. Which I don't want.
Brenner COP doesn't matter whether it's on or off if his COU is dead, since there's no zone and the only "real" COP effect is the instant heal.
Waylon COP matters because it has a defense boost that isn't tied to the zone explicitly. So I added this code to turn it off when he dies.
> I don't see the difference between Waylon's ability and say, Gage's (whose `moveMod` grants indirect range - this appears to be a common unit modifier variable name regardless of effect).

Good catch, that is bad pasta and should be fixed.
Gage's indirect range doesn't impact combat when it's not his turn, so just turning his CO zone off when his COU dies is technically enough (assuming you don't step on your own COU with a Kaiju, I guess).
It would look weird that he had the "power active" effect when no stat boosts are in play, but mechanically it'd be the same as if his power was turned off.
By contrast, Waylon's stat boost will affect combat when it's not his turn.

Reply:
> Yes. It does that because otherwise Waylon's ability would still work after he dies. Which I don't want.

Oh. Okay. I interpreted your initial statement "Waylon's powers specifically do actually matter while his COU is dead" to mean that his stat boosts _should_ outlive his CO unit. Now I get it: this is the correct behavior for all DoR COs, but you could technically have gotten away with not doing it, if it wasn't for that meddling Waylon.